tests/test_rate_limiter.py

```
@@ -0,0 +1,87 @@
+
+    assert result == "Success"
+    assert mock_func.await_count == 2
+
+
+@pytest.mark.asyncio
+async def test_max_retries_exceeded():
+    limiter = DiscordRateLimiter(redis_url=None, max_retries=2)
+
+    mock_func = AsyncMock(side_effect=Mock429())
+
+    with pytest.raises(Exception):
```

Ruff (0.15.1)

[warning] Do not assert blind exception: `Exception`

(B017)